packages/bot/src/stores/scratch-store.js

```diff
@@ -20,6 +23,27 @@ class ScratchStore {
     static get instance() {
         return this.singleton;
     }
+
+    registerReactions() {
+        // Syncs all trade options blocks' currency with the client's active currency.
+        this.disposeCurrencyReaction = reaction(
+            () => this.root_store.core.client.currency,
+            (currency) => {
+                const workspace = Blockly.derivWorkspace;
+
+                if (workspace) {
+                    const trade_options_blocks = workspace.getAllBlocks().filter(b => b.type === 'trade_definition_tradeoptions');
```

Comment:
@aaron-binary can you please define a `setCurrency` in scratch helper and call it from here. lets keep all the scratch related functions in scratch.

Reply:
The `setCurrency` function is already defined on the block object, I think creating another function that would update all trade options blocks' currencies would be overkill as it will not be re-used anywhere else.

Reply:
The `setCurrency` function is already defined on the block object, I think creating another function that would update all trade options blocks' currencies would be overkill as it will not be re-used anywhere else.